### PR TITLE
add: `qtile-git`

### PR DESCRIPTION
**Pull request by Ruturajn**
- Adding qtile to the list of supported packages.
- ```qtile-git.pacscript``` has been added.
- The ```packagelist``` is also updated with qtile-git.

### CODE REVIEW

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:3]: You don't need `version` here

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:8]: Add `pkgver`

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:10]: You don't need to cd here

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:11]: This step is done by pacstall automatically

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:12]: Not needed, pacstall does this automatically

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:29]: Remove this

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:31]: `sudo echo` is a useless statement, use `echo` instead

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:36]: Redirect to `/dev/null`

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:37]: Remove this

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:38]: Don't touch anything in the users home directory

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:39]: Remove this

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:40]: Remove this

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:42]: Remove this

**Review by Henryws — CHANGES_REQUESTED** [review]

**Comment by Ruturajn** [comment]
Thanks, I will make the suggested changes.

**Comment by Ruturajn** [comment]
I have made all the changes. Just one thing, pacstall is giving me a warning about the package maintenance, so do I need to add myself as the maintainer?

**Comment by wizard-28** [comment]
> I have made all the changes. Just one thing, pacstall is giving me a warning about the package maintenance, so do I need to add myself as the maintainer? So, once I know this I will commit the changes.
> 
> Thanks

Yes add yourself as the maintainer.

**Comment by Ruturajn** [comment]
I have made all the changes. Kindly review it.

Thanks

**Review by wizard-28 — CHANGES_REQUESTED** [review]
Add `removescript()` function, to remove it.

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:38]: This should be done in `install`

**Review by Henryws — CHANGES_REQUESTED** [review]

**Comment by Ruturajn** [comment]
I have added the commands to the ```install()``` function as suggested.

File: packages/qtile-git/qtile-git.pacscript
Comment (Henryws) [inline packages/qtile-git/qtile-git.pacscript:29]: This should be `/usr/src/pacstall/qtile-git/usr/share/xessions/qtile.desktop`

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/qtile-git/qtile-git.pacscript
Comment (wizard-28) [reply packages/qtile-git/qtile-git.pacscript:29]: Ooh. I remember I had to do similar thing in [awesome-git](https://github.com/pacstall/pacstall-programs/blob/master/packages/awesome-git/awesome-git.pacscript) too

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:29]: Never mind, it's showing up now. I will test it again though and then commit.

**Comment by Ruturajn** [comment]
Yeah, the qtile session shows up. I have made the changes. Please review them.

Thanks

File: packages/qtile-git/qtile-git.pacscript
Comment (wizard-28) [inline packages/qtile-git/qtile-git.pacscript:38]:
```suggestion
```

Not required.

**Review by wizard-28 — CHANGES_REQUESTED** [review]
You might also want a [`repology`](https://github.com/pacstall/pacstall/wiki/Pacscript-101#repology) variable in the pacscript to have an easier experience updating it in the future.

File: packages/qtile-git/qtile-git.pacscript
Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:38]: Ok Thanks

**Comment by Ruturajn** [comment]
Ok Thanks

File: packages/qtile-git/qtile-git.pacscript
Comment (wizard-28) [inline packages/qtile-git/qtile-git.pacscript:33]: @Henryws shouldn't this be a separate file though?

File: packages/qtile-git/qtile-git.pacscript
Comment (wizard-28) [inline packages/qtile-git/qtile-git.pacscript:39]: @Henryws is this really needed?

**Comment by Ruturajn** [comment]
Are there any other changes that I need to make?

File: packages/qtile-git/qtile-git.pacscript
Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I added it since it remains after I remove the package, and you suggested that I add it.

File: packages/qtile-git/qtile-git.pacscript
Comment (wizard-28) [reply packages/qtile-git/qtile-git.pacscript:33]: Henryws: https://discord.com/channels/839818021207801878/866673569521991700/956582537273176105

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:33]: I am not exactly sure, what you mean here

Comment (wizard-28) [reply packages/qtile-git/qtile-git.pacscript:33]: > I am not exactly sure, what you mean here

@Henryws replied on discord that it's fine to have it embedded in the pacscript. That's the discord link to the message for archival purposes.

File: packages/qtile-git/qtile-git.pacscript
Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: image

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: This is the output from ```pacstall -T qtile-git```

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I think that the file ```qtile.desktop``` is being created in ```/usr/share/xsessions/``` automatically, without explicitly doing it in the pacscript.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: What do I need to change then?

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Yes, because of a previous build

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: So, should I remove the ```removescript()``` function?

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Figure out how to build it to the proper location first.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Where should it be ideally?

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Ok thanks, I will try it out this way, and about the directory should it be the ```$STOWDIR```?

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: `${STOWDIR}/${pkgname}`

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Ok

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I made the following changes,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo pip3 install --target=/usr/src/pacstall/qtile-git .
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git install
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
But, I am getting this error:
```
Finished processing dependencies for qtile==0.21.0
[+] INFO: Symlinking files
WARNING! stowing qtile-git would cause conflicts:
  * existing target is not owned by stow: bin
All operations aborted.
[!] ERROR: Package contains links to files that exist on the system
[+] INFO: Cleaning up
[!] ERROR: Failed to install qtile-git

```

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Ok

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: [qtile-test.txt](https://github.com/pacstall/pacstall-programs/files/8347217/qtile-test.txt)

I have directed the output of ```tree /tmp/qtile-git/``` to this file.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Even after passing the target directories to setup.py and pip, ```/usr/local/bin/qtile```, is being created.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I changed the order for ```setup.py``` and ```pip```,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git install
  sudo pip3 install --target=/usr/src/pacstall/qtile-git/ . 
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
Now, there is no ```/usr/local/bin/qtile``` file created, but the ```stow``` error remains.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I am doing it like this now,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git/usr install
  sudo pip3 install . --target=/usr/src/pacstall/qtile-git/usr/ 
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
And now I get this,
Screenshot from 2022-03-25 11-47-43

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I changed the ```install()``` function to this,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo pip3 install . --target=/usr/src/pacstall/qtile-git/usr
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```

I checked the Qtile Documentation, where it's mentioned that only ```pip install``` needs to be done after cloning the git repository : http://docs.qtile.org/en/stable/manual/install/index.html#installing-from-source

And, now only ```/usr/bin/qtile``` is created
Screenshot from 2022-03-25 14-07-37

This is what was expected right?

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I have changed the ```install()``` function to this,
```
install() {
  sudo pip3 install --upgrade --no-cache --force-reinstall cairocffi cffi xcffib
  sudo pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git/usr install --root=/usr/src/pacstall/qtile-git/
  sudo cp ./resources/qtile.desktop /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop
}
```
The ```qtile.desktop``` file is already present in qtile's git repo, so I copied it from there, and ```setup.py install``` and ```pip install .``` do the basically the same thing, and since using ```setup.py install``` allows for easier control over the target installation directory, I have used that.





This is after ```pacstall -R qtile-git```,



So, should I commit this?

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Run `pacstall -R qtile-git` with your fix

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Yeah, I have attached the screenshot for it. Should I attach the whole output?

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Send a file

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Oh, yeah btw I removed the ```removescript()``` function since, ```pacstall -R``` was handling everything

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: [qtile-test-out.txt](https://github.com/pacstall/pacstall-programs/files/8352832/qtile-test-out.txt)

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: I meant the output of `pacstall -R qtile-git`

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Yeah I have directed the output from that command to this file

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Oh, sorry!! I meant `-T`

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: ohh ok

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: [qtile-pacstall-T-qtile-git.txt](https://github.com/pacstall/pacstall-programs/files/8352876/qtile-pacstall-T-qtile-git.txt)

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Running your pacscript returns this with `pacstall -T qtile-git`
```bash
/
`-- usr
    `-- share
        `-- xsessions
            `-- qtile.desktop

3 directories, 1 file
```

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I think you have used the original pacscript which did not have the fix that I have provided above. Could you try it out with this : [qtile-git-new-pacscript.txt](https://github.com/pacstall/pacstall-programs/files/8354605/qtile-git-new-pacscript.txt)

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Ok, it worked in the docker, testing on my main machine. 🤞🏻

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Yeah, I am thinking about how to fix that

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Is this fine ?
[pacstall-T-qtile-git-new.txt](https://github.com/pacstall/pacstall-programs/files/8355795/pacstall-T-qtile-git-new.txt)

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Send me the new pacscript. I can't test your fix without it.

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: I removed the build command in the ```setup.py install```

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: image
this line?

Comment (Ruturajn) [reply packages/qtile-git/qtile-git.pacscript:39]: Yeah

Comment (Henryws) [reply packages/qtile-git/qtile-git.pacscript:39]: Cool that works

**Comment by Ruturajn** [comment]
Should I push these changes then?

**Comment by Henryws** [comment]
CI is failing, but it works on my computer

**Comment by Ruturajn** [comment]
I have tested this in multiple virtual machines

**Review by Henryws — APPROVED** [review]

**Comment by wizard-28** [comment]
Finally after 83 conversations lets merge this! 🎉

**Comment by Ruturajn** [comment]
Thank you guys, for all the support. This was my first open-source contribution, I learnt a lot.

**Pull request merged**
